widget/simple_renderer.go

```diff
@@ -0,0 +1,57 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+)
+
+// newSimpleRenderer creates a new simpleRenderer.
+func newSimpleRenderer(widget SimpleWidget, objects []fyne.CanvasObject, layout func(size fyne.Size)) *simpleRenderer {
+	return &simpleRenderer{
+		widget:  widget,
+		objects: objects,
+		layout:  layout,
+	}
+}
+
+// simpleRenderer is a renderer providing the basic rendering functionality used
+// by SimpleWidgetBase.
+type simpleRenderer struct {
+	widget  fyne.Widget
+	objects []fyne.CanvasObject
+
+	layout func(size fyne.Size)
+}
+
+// MinSize returns the minimum size of the widget that is rendered by this renderer.
+func (s *simpleRenderer) MinSize() fyne.Size {
+	return fyne.Size{Width: 0, Height: 0}
+}
```

Comment:
Do we need to be able to overwrite the MinSize function in the renderer, if it is already overwritten in the widget? If yes, I can make the renderer return the same as the MinSize function of the widget if overwritten.

Reply:
@andydotxyz?

Reply:
MinSize can now be overwritten in widget and will be passed down to the renderer.

Reply:
Most widgets have a min size defined by their renderring - and so the default widget `MinSize` implementation simply delegates to the renderer.
Remember that we want complete separation between widget and render output (in the API design).